### PR TITLE
Macroserver Environment Variables Catalog

### PR DESCRIPTION
**Pull request by amilan**
This PR is covering the creation of a MacroServer Environment Variables Catalog.

### CODE REVIEW

**Comment by amilan** [comment]
In theory this PR was also going to include the creation of the macro environment documentation, but we can keep this one for the catalog only and create a new one with the macro environment documentation.

File: doc/source/users/environment_variable_catalog.rst [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+Environment Variable Catalog
+============================
+
+This is the catalog of the available environment variables that can be used in
+Sardana.
+
+.. toctree::
+    :maxdepth: 3
+
+    environment_variable_catalog
```
Comment (daneos) [inline doc/source/users/environment_variable_catalog.rst]: This causes `circular toctree references` warnings.

File: doc/source/users/environment_variable_catalog.rst [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+.. note:: You can use :ref:`expconf <expconf_ui>` to set up pre-scan snapshot.
+
+.. _sampleinfo:
+
+SampleInfo
+~~~~~~~~~~
+    *Not mandatory, set by user*
+
+Extra info for the sample. describe la muestra.
```
Comment (daneos) [inline doc/source/users/environment_variable_catalog.rst]: Should be better explained, possibly not in Spanish ;)

File: doc/source/users/environment_variable_catalog.rst [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+.. seealso:: For more information about the implementation details of the scan
+             macros in Sardana, see 
+             :ref:`scan framework <sardana-macros-scanframework>`
+
+.. _sourceinfo:
+
+SourceInfo
+~~~~~~~~~~
+*Not mandatory, set by user*
+
```
Comment (daneos) [inline doc/source/users/environment_variable_catalog.rst]: Missing documentation

File: doc/source/users/macro_environment.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+.. _macro-environment:
+
+=================
+Macro Environment
+=================
+
```
Comment (daneos) [inline doc/source/users/macro_environment.rst]: To be done in separate PR

File: doc/source/users/scan.rst [hunk 5]
```diff
@@ -183,91 +183,9 @@ Scans are highly configurable using the environment variables
...
-
-
-.. seealso:: For more information about the implementation details of the scan
-             macros in Sardana, see 
-             :ref:`scan framework <sardana-macros-scanframework>`
+.. seealso:: For further information about the available Sardana Environment
+             Variables, check the 
+             :ref:`Environment Variable Catalog <environment-variable-catalog>`
 
 .. _zero order hold: https://en.wikipedia.org/wiki/Zero-order_hold
```
Comment (daneos) [inline doc/source/users/scan.rst]: These links should probably be moved to the env var catalog file for Sphinx to see them.

File: doc/source/users/environment_variable_catalog.rst (hunk 2 above)
Comment (amilan) [reply doc/source/users/environment_variable_catalog.rst]: Uuuupppsss ... that shouldn't be in this PR. I'm going to fix it.

File: doc/source/users/environment_variable_catalog.rst (hunk 3 above)
Comment (amilan) [reply doc/source/users/environment_variable_catalog.rst]: Same as before ...

File: doc/source/users/scan.rst (hunk 5 above)
Comment (amilan) [reply doc/source/users/scan.rst]: 👍

File: doc/source/users/environment_variable_catalog.rst (hunk 3 above)
Comment (amilan) [reply doc/source/users/environment_variable_catalog.rst]: Fixed. The explanation is not the most explicit one, but could be enough at this moment.

File: doc/source/users/macro_environment.rst (hunk 4 above)
Comment (amilan) [reply doc/source/users/macro_environment.rst]: Maybe we can keep this file already here and I will create a new task for filling this file.

File: doc/source/users/environment_variable_catalog.rst (hunk 1 above)
Comment (amilan) [reply doc/source/users/environment_variable_catalog.rst]: I'm trying to find a solution for this case ...

**Comment by daneos** [comment]
Fixes #229 and #633
